compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -627,19 +627,25 @@ execPackage projectOpts filePath opts mbOutFile dalfInput =
 
     targetFilePath = fromMaybe defaultDarFile mbOutFile
 
+-- | Given a path to a .dalf or a .dar return the bytes of either the .dalf file
+-- or the the main dalf from the .dar
+-- In addition to the bytes, we also return the basename of the dalf file.
+getDalfBytes :: FilePath -> IO (B.ByteString, FilePath)
+getDalfBytes fp
+  | "dar" `isExtensionOf` fp = do
+        dar <- B.readFile fp
+        let archive = ZipArchive.toArchive $ BSL.fromStrict dar
+        manifest <- either fail pure $ readDalfManifest archive
+        dalfs <- either fail pure $ readDalfs archive
```

Comment:
Surprised `either fail pure` isn't in extra. It's `requiredE` in the assistant codebase (± exception handling details), but that is neither here nor there.